app/javascript/components/batch_update/components/TextInput.js

```diff
@@ -10,12 +10,25 @@ class TextInput extends React.Component {
     }
   }
 
+  handleKeyDown = e => {
+    if (!this.props.numeric) return
+    if (e.metaKey) return
+    if (e.ctrlKey) return
+    if (e.key === 'Enter') return
+    if (e.key === 'Backspace') return
+    if (e.key === 'Delete') return
+    if (e.key === 'Tab') return
+    if (e.keyCode >= 48 && e.keyCode <= 57 && !e.shiftKey) return
```

Comment:
Sorry it's not obvious from reading this — what's the special key-handling that's required here?

Reply:
We want to prevent any non-numeric inputs from being entered. I could have used `<input type='number'>` but the semantics of that felt a little bit weird to me and this way I could just reuse an existing input.

Basically I'm saying: allow enter, backspace, delete and tab. allow numbers. allow meta+key combinations. Do not allow shift+numbers. But I agree that this feels like a strange way to enforce this.

Reply:
Ah I see, makes sense. So one more thing. Maybe edge-case-y, but might as well allow decimal point here `.` for numerics, yeah?